hw-2/controllers/users.ts

```diff
@@ -0,0 +1,47 @@
+import { Request, Response } from 'express';
+import { ValidatedRequest } from 'express-joi-validation';
+import UsersStorage from '../storage/users';
+import { UserRequestSchema, User } from '../models/user';
+
+const usersStorage = new UsersStorage();
+
+function prepareUser(user: User | null): User | null {
+  if (user === null) return null;
+
+  const preparedUser = {
+    ...user,
+  };
+  delete preparedUser.isDeleted;
+
+  return preparedUser;
+}
+
+export const getUsers = (req: Request, res: Response): void => {
+  const users = usersStorage.getUsers(
+    req.query.login, Number(req.query.limit) || undefined,
+  ).map((user) => prepareUser(user));
+  res.json(users);
+};
+
+export const getUser = (req: Request, res: Response): void => {
+  res.json(prepareUser(usersStorage.getUserById(req.params.id)));
```

Comment:
When we create intermediary variables, the code is easier to read:

```typescript
export const getUser = ({ params }: Request, res: Response): void => {
  const user = usersStorage.getUserById(params.id);
  const preparedUser = prepareUser(user);

  res.json(preparedUser);
};
```

Reply:
Got it! Thank you again!